### DISCUSSION TITLE
Confusion regarding tensor.dtypes

### DISCUSSION THREAD

**Discussion by AksharGoyal** [mrdbourke/pytorch-deep-learning#605] [Q&A] [answered]
## Course video: Learn PyTorch for deep learning in a day. Literally., timestamp: 1:57:00

Hello, I was experimenting with the dtypes and I got this result. I am curious to understand what's the logic behind this. Why am I only getting 4 digits after 0 and why not more after increasing precision?

## My code

```python
torch.tensor([0.0102030405], dtype=torch.float16),
torch.tensor([0.0102030405], dtype=torch.float32),
torch.tensor([0.0102030405], dtype=torch.float64)
```

<img width="641" alt="image" src="https://github.com/mrdbourke/pytorch-deep-learning/assets/38995624/520fbad7-fc99-448e-adfa-99e62c74df47">

**Comment by mrdbourke** [comment, marked as answer]
Hi there, 

This is likely due to setting `torch.set_printoptions`, see: https://pytorch.org/docs/stable/generated/torch.set_printoptions.html#torch.set_printoptions 

And see: https://discuss.pytorch.org/t/how-to-print-more-than-4-digits-after-decimal/63405 

## Example

Default print options.

```python
import torch

float16 = torch.tensor([0.0102030405], dtype=torch.float16)
float32 = torch.tensor([0.0102030405], dtype=torch.float32)
float64 = torch.tensor([0.0102030405], dtype=torch.float64)

print(float16)
print(float32)
print(float64)
```

Output:

```
tensor([0.0102], dtype=torch.float16)
tensor([0.0102])
tensor([0.0102], dtype=torch.float64)
```

Set print options precision to 10:

```python
# Set precision equal to 10
torch.set_printoptions(precision=10)

print(float16)
print(float32)
print(float64)
```

Output:

```
tensor([0.0102005005], dtype=torch.float16)
tensor([0.0102030402])
tensor([0.0102030405], dtype=torch.float64)
```

**Reply by AksharGoyal** [reply]
This is helpful, thank you so much!
